### PR TITLE
Updated based on feedback

### PR DESCRIPTION
**Pull request by stenio123**
Update based on https://github.com/cloud-native-security-controls/controls-catalog/issues/6

Detailed feedback response: https://drive.google.com/file/d/1zTmlpXZezjoQJrA7LzYIjO2yFen72SsJ/view?usp=sharing

### CODE REVIEW

**Comment by stenio123** [comment]
@JonZeolla JonZeolla

**Comment by JonZeolla** [comment]
@pratiklotia do you have an idea of when you'll get a chance to look at this?

File: controls/controls_catalog_v1.0.csv [hunk 1]
```diff
@@ -1,233 +1,233 @@
...
-178,SSCP v1.0,Securing Materials,Generate an immutable SBOM of the code,There are currently two well known SBOM specifications: SPDX34 and CycloneDX,,Moderate to High,Moderate to High
+SC-17 PUBLIC KEY INFRASTRUCTURE CERTIFICATES",High,High
+170,SSCP v1.0,Securing Deployments,Ensure clients can perform verification of artefacts and associated metadata,,"SI-7 SOFTWARE, FIRMWARE, AND INFORMATION INTEGRITY",Moderate to High,Moderate to High
+171,SSCP v1.0,Securing Deployments,Ensure clients can verify the “freshness” of files,Ensure clients can access latest versions and can veriify if the provided files are out of date,"SI-7 SOFTWARE, FIRMWARE, AND INFORMATION INTEGRITY",Moderate to High,Moderate to High
+172,SSCP v1.0,Securing Deployments,Use a framework for managing software updates,"Use TUF to enforce the updating of software. The Update Framework (TUF) is a specification for delivering software updates in a secure, reliable and trusted way",,High,High
```
Comment (JonZeolla) [inline controls/controls_catalog_v1.0.csv:207]: There was some commentary on this item; changed this to 'consider' and accept the upstream feedback suggestion.
```suggestion
172,SSCP v1.0,Securing Deployments,Use an automated approach for managing software updates,"Consider using The Update Framework (TUF) to enforce the updating of software. TUF is a specification for delivering software updates in a secure, reliable and trusted way",,High,High
```

File: controls/controls_catalog_v1.0.csv [hunk 2]
```diff
@@ -1,233 +1,233 @@
...
+190,SSCP v1.0,Securing the Source Code,Use branch protection rules,"SCM platforms allow the configuration and restriction of source code operations on individual branches. Protection rules can be used to enforce the usage of pull requests with specified precondition and approval rules, ensuring that a human code review process is followed or an automated status checking of a branch occurs. Additionally, protected branches can be used to disallow dangerous use of force pushes26, preventing the overwrite of commit histories and potential obfuscation of code changes.",SA-8 SECURITY ENGINEERING PRINCIPLES,Moderate to High,Moderate to High
+191,SSCP v1.0,Securing the Source Code,Enforce MFA for accessing source code repositories,,IA-2(1) Identification and Authentication (organizational Users) | Multi-Factor Authenticaiton to Priviledged Accounts,Moderate to High,Moderate to High
+192,SSCP v1.0,Securing the Source Code,Use SSH keys to provide developers access to source code repositories,,AC-1 REMOTE ACCESS,Moderate to High,Moderate to High
```
Comment (JonZeolla) [inline controls/controls_catalog_v1.0.csv:231]: The contradiction of 191 and 192 is great feedback for the upstream software supply chain white paper.

**Review by JonZeolla — CHANGES_REQUESTED** [review]
Feedback inline, otherwise looking great!  Didn't check that this properly renders as CSV afterward my suggested changes.

**Comment by JonZeolla** [comment]
@stenio123 @pratiklotia FYI I did an initial review of this

File: controls/controls_catalog_v1.0.csv [hunk 3]
```diff
@@ -1,233 +1,233 @@
...
+ID,Originating Document,Section,Control Title,Control Implementation,NIST SP800-53r5 references,Assurance Level,Risk Categories
+1,CNSWP v1.0,Access,"Secrets are injected at runtime, either as environment variables or as a file",,IA-5(7) Authenticator Management | No Embedded Unencrypted Static Authenticators,N/A,N/A
+2,CNSWP v1.0,Access,Applications and workloads are explicitly authorized to communicate with each other using mutual authentication,,IA-9 Service Identification and Authentication,N/A,N/A
+3,CNSWP v1.0,Access,Keys are rotated frequently,,SC-12 Cryptographic Key Establishment and Management,N/A,N/A
+4,CNSWP v1.0,Access,Key lifespan is short,,SC-12(3) Cryptographic Key Establishment and Management | Asymetric Key,N/A,N/A
+5,CNSWP v1.0,Access,Credentials and keys protecting sensitive workloads (health/finance/etc) are generated and managed independent of a cloud service provider,KMS and HMS are common technologies to achive this. FIPS 140-2 complaince is strongly suggested. Cloud KMS tends to be FIPS 140-2 Level 2 or greater.,IA-2(12) Identification and Authentication (Organizational Users) | Acceptance of PIV Credentials,N/A,N/A
```
[the remark this answers is not in the document]
Comment (pratiklotia) [reply controls/controls_catalog_v1.0.csv:6]: Looks like these typos existed even before @stenio123' PR. (My understanding is that this line has been copied as is). I'd leave it up to @stenio123 if he would like to address these changes or have it in a separate PR.

File: controls/controls_catalog_v1.0.csv [hunk 4]
```diff
@@ -1,233 +1,233 @@
...
+25,CNSWP v1.0,Compute,API auditing is enabled with a filter for a specific set of API Groups or verbs,"API audits of the application, kubernetes API server, and kernel should be implemented.",AU-2 Event Logging,N/A,N/A
+26,CNSWP v1.0,Compute,Container specific operating systems are in use,a read-only OS with other services disabled. This provides isolation and resource confinement that enables developers to run isolated applications on a shared host kernel,CM-2 Baseline Configuration CM-7 Least Functionality,N/A,N/A
+27,CNSWP v1.0,Compute,The hardware root of trust is based in a Trusted Platform Module (TPM) or virtual TPM (vTPM),"Ensure HW root of trust extends to the OS kernel, modules, system images, container runtimes, and all software on the system.","SI-7 Software, Firmware, and Information Integrity",N/A,N/A
```
[the remark this answers is not in the document]
Comment (pratiklotia) [reply controls/controls_catalog_v1.0.csv:30]: Up to @stenio123 to take this suggestion. (Pre-existing)

File: controls/controls_catalog_v1.0.csv [hunk 5]
```diff
@@ -1,233 +1,233 @@
...
+43,CNSWP v1.0,Compute,"Actionable audit events are generated that correlate/contextualize data from logs into ""information"" that can drive decision trees/incident response",,AU-3 Content of Audit Records,N/A,N/A
+44,CNSWP v1.0,Compute,Segregation of duties and the principle of least privilege is enforced,,AC-6 Least Privilege,N/A,N/A
+45,CNSWP v1.0,Compute,Non-compliant violations are detected based on a pre-configured set of rules defined by on the organization's policies,,"SI-7 Software, Firmware, and Information Integrity",N/A,N/A
+46,CNSWP v1.0,Compute,Native secret stores encrypt with keys from an external Key Management Store (KMS),,SC-12(3) Systems & Communication Protection,N/A,N/A
+47,CNSWP v1.0,Compute,Native secret stores are not configured for base64 encoding or stored in clear-text in the key-value store by default,Encoding is not encryption,SC-12(3) Systems & Communication Protection,N/A,N/A
```
[the remark this answers is not in the document]
Comment (pratiklotia) [reply controls/controls_catalog_v1.0.csv:51]: @JonZeolla Wouldn't the line above be a sufficient pointer? I felt that that control is specifically written to address what not to use.

File: controls/controls_catalog_v1.0.csv [hunk 6]
```diff
@@ -1,233 +1,233 @@
...
+61,CNSWP v1.0,Deploy,Forensics capabilities are integrated into an incident response plan and procedures,,INCIDENT HANDLING | MALICIOUS CODE AND FORENSIC ANALYSIS,N/A,N/A
+62,CNSWP v1.0,Deploy,"AI, ML, or statistical modeling are used for behavioural and heuristic environment analysis",,SI-3 SYSTEM AND INFORMATION INTEGRITY,N/A,N/A
+63,CNSWP v1.0,Develop,Establish a dedicated Production environment,"Ensure that production workloads are in a separate, dedicated environment from non-production workloads. In the context of containers, this can mean separate clusters. In the case of VMs, separate networks.",SA-3(1) SYSTEM DEVELOPMENT LIFE CYCLE | MANAGE PREPRODUCTION ENVIRONMENT,N/A,N/A
+64,CNSWP v1.0,Develop,Leverage Dynamic deployments,"Blue/Green, Alpha/Beta, Canary, red-black deployments",SA-8(31) SECURITY AND PRIVACY ENGINEERING PRINCIPLES | SECURE SYSTEM MODIFICATION,N/A,N/A
+65,CNSWP v1.0,Develop,Integrate vulnerability and configuration scanning in either the IDE or at the CI system during pull request,,SA-11(1) DEVELOPER TESTING AND EVALUATION | STATIC CODE ANALYSIS,N/A,N/A
```
Comment (pratiklotia) [inline controls/controls_catalog_v1.0.csv:71]: Suggestion: `vulnerability and configuration scanning in different stages of development such as the IDE and at the CI system during pull request`
I felt like IDE and CI scanning are not necessarily complementary. Both can and should co-exist.

File: controls/controls_catalog_v1.0.csv [hunk 7]
```diff
@@ -1,233 +1,233 @@
...
-108,CNSWP v1.0,Distribute,HSMs or credential managers should be used for protecting credentials,,SC-12(3) SYSTEMS & COMMUNICATION PROTECTION,N/A,N/A
-109,CNSWP v1.0,Distribute,Container image scanning findings are acted upon,,SI-2(3) SYSTEM & INFORMATION INTEGRITY,N/A,N/A
-110,CNSWP v1.0,Distribute,organizational compliance rules are enforced,,PL-1 POLICY AND PROCEDURES,N/A,N/A
-111,CNSWP v1.0,Distribute,Incremental hardening of the infrastructure is employed,,,N/A,N/A
-112,CNSWP v1.0,Distribute,pulls from public registries are controlled and only from authorized engineers or internal registries,,AC-6(3) LEAST PRIVILEGE | NETWORK ACCESS TO PRIVILEGED COMMANDS,N/A,N/A
+SA-3 SYSTEM DEVELOPMENT LIFE CYCLE",N/A,N/A
+106,CNSWP v1.0,Distribute,Enable image signing key revokation in the event of compromise,,SI-7 SYSTEM & INFORMATION INTEGRITY,N/A,N/A
+107,CNSWP v1.0,Distribute,Security updates are prioritized,,SI-2(3) SYSTEM & INFORMATION INTEGRITY,N/A,N/A
+108,CNSWP v1.0,Distribute,HSMs or credential managers should be used for protecting credentials,,SC-12(3) SYSTEMS & COMMUNICATION PROTECTION,N/A,N/A
```
[the remark this answers is not in the document]
Comment (pratiklotia) [reply controls/controls_catalog_v1.0.csv:117]: Wouldn't `credential managers` in the first line itself mean software based ones? Alternatively, `HSMs or software-based credential managers should be used for protecting credentials`

**Review by pratiklotia — CHANGES_REQUESTED** [review]
Added minor suggestions, rest of the changes look good to me.
I'm concerned why the `file changed` tab shows so many changes. Has the formatting of the doc changed? Ideally only the controls where you have made changes should have shown up there. Instead it shows almost lines being changed.

**Comment by JonZeolla** [comment]
@pratiklotia yes I think it is because of formatting.  I did a fair amount of manual validation of the content

File: controls/controls_catalog_v1.0.csv (hunk 7 above)
[the remark this answers is not in the document]
Comment (JonZeolla) [reply controls/controls_catalog_v1.0.csv:117]: I'm good either way

File: controls/controls_catalog_v1.0.csv (hunk 5 above)
[the remark this answers is not in the document]
Comment (JonZeolla) [reply controls/controls_catalog_v1.0.csv:51]: I was thinking algorithms, but also wouldn't reinvent the wheel.  Instead just point to other docs like what [OWASP has](https://cheatsheetseries.owasp.org/cheatsheets/Cryptographic_Storage_Cheat_Sheet.html).

File: controls/controls_catalog_v1.0.csv (hunk 3 above)
[the remark this answers is not in the document]
Comment (JonZeolla) [reply controls/controls_catalog_v1.0.csv:6]: There was feedback on this item in [here](https://drive.google.com/file/d/1zTmlpXZezjoQJrA7LzYIjO2yFen72SsJ/view) which talks about the customer managed recommendation

**Comment by JonZeolla** [comment]
@stenio123 do you know when you'll have a chance to look at this again?  Thanks!

**Comment by JonZeolla** [comment]
@stenio123 I'd love to get this merged, have you had a chance to look at the feedback?

File: controls/controls_catalog_v1.0.csv [hunk 8]
```diff
@@ -1,233 +1,234 @@
...
+61,CNSWP v1.0,Deploy,Forensics capabilities are integrated into an incident response plan and procedures,,INCIDENT HANDLING | MALICIOUS CODE AND FORENSIC ANALYSIS,N/A,N/A
+62,CNSWP v1.0,Deploy,"AI, ML, or statistical modeling are used for behavioural and heuristic environment analysis to detect unwanted activities",,SI-3 SYSTEM AND INFORMATION INTEGRITY,N/A,N/A
+63,CNSWP v1.0,Develop,Establish a dedicated Production environment,"Ensure that production workloads are in a separate, dedicated environment from non-production workloads. In the context of containers, this can mean separate clusters. In the case of VMs, separate networks.",SA-3(1) SYSTEM DEVELOPMENT LIFE CYCLE | MANAGE PREPRODUCTION ENVIRONMENT,N/A,N/A
+64,CNSWP v1.0,Develop,Leverage Dynamic deployments,"Blue/Green, Alpha/Beta, Canary, red-black deployments",SA-8(31) SECURITY AND PRIVACY ENGINEERING PRINCIPLES | SECURE SYSTEM MODIFICATION,N/A,N/A
+65,CNSWP v1.0,Develop,Integrate vulnerability and configuration scanning in either the IDE or at the CI system during pull request,,SA-11(1) DEVELOPER TESTING AND EVALUATION | STATIC CODE ANALYSIS,N/A,N/A
```
Comment (pratiklotia) [inline controls/controls_catalog_v1.0.csv:72]:
```suggestion
65,CNSWP v1.0,Develop,Integrate vulnerability and configuration scanning in both the IDE and at the CI system during pull request,,SA-11(1) DEVELOPER TESTING AND EVALUATION | STATIC CODE ANALYSIS,N/A,N/A

```

**Review by JonZeolla — APPROVED** [review]
LGTM

**Pull request merged**